### ISSUE TITLE
Dimension ordering is not preserved when defined

### ISSUE THREAD

**Issue by jxrossel** [ioam/holoviews#1383]
Hi,

According to the docs (http://holoviews.org/Tutorials/Elements.html >> Bars), 

> To preserve the initial ordering specify the Dimension with values set to 'initial', or you can supply an explicit list of valid dimension keys

In `dimension.Dimension.values`, the doc says:

> Optional specification of the allowed value set for the dimension that may also be used to retain a categorical  ordering

But the code reads:
```
        values = params.get('values', [])
        if isinstance(values, basestring) and values == 'initial':
            self.warning("The 'initial' string for dimension values is no longer supported.")
            values = []
        all_params['values'] = sorted(list(unique_array(values)))
```

So that (1) 'initial' cannot be used, (2) ordering is not preserved (thanks to the call to `sorted`)

**Comment by philippjfr** [comment]
Thanks for catching this, we need to update the tutorial, ``initial`` is no longer supported in general. Also @jlstevens do you know why we're sorting the ``values`` here?

**Comment by jlstevens** [comment]
I think this was something @basnijholt requested for ordered sliders.

**Comment by jlstevens** [comment]
I think making the values unique is sensible, not sure about the sorting now we removed ``'initial'``...

**Comment by jlstevens** [comment]
Whatever the fix is, I'm sure it is simple. Philipp and I just need to make sure the semantics are clear and that we aren't breaking backwards compatibility.

**Comment by jxrossel** [comment]
Well, I guess that if a user specifies the dimension values, they should be left as given (or possibly made unique, but that could be the user's responsability as well)

**Comment by jlstevens** [comment]
I think automatically making them unique is almost always appropriate and I don't *think* anything is relying on the sorting (which I agree we should remove). Any thoughts @philippjfr ?

**Comment by philippjfr** [comment]
>I think automatically making them unique is almost always appropriate and I don't think anything is relying on the sorting (which I agree we should remove). Any thoughts @philippjfr ?

I think the original issue was related to partial overlaps, i.e. where one DynamicMap defines some list of values and another defines a different partially overlapping list of values. In that case sorting is usually the most sensible thing to do, but I don't think it's the best policy in general.

**Comment by jlstevens** [comment]
Ok, sounds like we should just disable the sorting. I'm happy to do that.

**Comment by jlstevens** [comment]
Addressed in the PR referenced above (#1457) - the values are now unique but not sorted automatically. Closing.
